### PR TITLE
Spin local dev environment with docker-compose

### PR DESCRIPTION
**Pull request by lioramilbaum**
Upgrade: neo4j, apoc, api parent image (fedora 32)

### CODE REVIEW

**Comment by lioramilbaum** [comment]
@mzidek-gh @hanzz take a look

File: docker-compose.yml
Comment (mprahl) [inline docker-compose.yml:19]: Will 3.5 work?

File: Makefile
Comment (mprahl) [inline Makefile:16]: Could we wrap this in tox so that dependencies are in a virtualenv and functional test and we can then call our additional tests such as linting and unit in the same way?

File: Makefile
Comment (mprahl) [inline Makefile:10]: I prefer it without the `-d` flag so you can see the logs. Any chance we can make that flag optional?

File: docker-compose.yml
Comment (mprahl) [inline docker-compose.yml:1]: Could you add a volume for Neo4j's database files?

File: .github/workflows/functional.yml
Comment (mprahl) [inline .github/workflows/functional.yml:1]: Very cool. Let me know what you need from me to enable this.

Comment (lioramilbaum) [reply .github/workflows/functional.yml:1]: GitHub Actions will start executing the workflow on future PRs, as soon as this one is merged.

File: Makefile
Comment (lioramilbaum) [reply Makefile:10]: We can have a different target for logs. Does that make sense?

File: Makefile
Comment (lioramilbaum) [reply Makefile:16]: Sure. Lets do that in the next PR. I don't want it to block this PR from been merged.

File: docker-compose.yml
Comment (lioramilbaum) [reply docker-compose.yml:19]: I think I've checked it and it didn't.  I think I've filed an issue for upgrading neo4j.

File: Makefile
Comment (mprahl) [reply Makefile:10]: Sure

**Review by mprahl — APPROVED** [review]

**Pull request merged**
